NEWS.md

```diff
@@ -439,6 +439,11 @@ Deprecated or removed
 
   * `cummin` and `cummax` have been deprecated in favor of `accumulate`.
 
+  * The idiosyncratic `Array` constructor syntax `Array(T, dims...)` has been deprecated
```

Comment:
Would you prefer different wording? For example "The `zeros(T, dims...)`/`ones(T, dims...)`-like `Array` constructor syntax `Array(T, dims...)`..."? Or something else? Best! :)